inst/tutorials/tidy-02-causality2/tidy-02-causality2.Rmd

```diff
@@ -0,0 +1,286 @@
+         answer("Generalizability"),
+         answer("All of these"))
+```
+
+## More Logicals in R
+
+### Complex relationals
+
+
+In this exercise, you have the ages of a sample of 15 people, stored in the `ages` vector. We can use these relational operators to create a logical vector which indicates which ages fall within a specific range. In particular, we can find out which respondents are college-aged (18-22).
+
+#### Exercise
+
+- Create a logical vector, called `college.aged`, which indicates which observations in `ages` are greater than or equal to 18 and less than or equal to 22. Be sure to use parentheses to separate out the two logical statements.
+- Take the sum of the `college.aged` vector to determine how many 18-22 year olds there are in the sample.
+
+```{r generate-ages}
+## check the value of the ages vector
+ages <- c(31, 20, 43, 45, 41, 46, 28, 49, 61, 19, 39)
+```
+
+
+```{r ages, exercise = TRUE, exercise.setup = "generate-ages"}
+## check the value of the ages vector
+ages <- c(31, 20, 43, 45, 41, 46, 28, 49, 61, 19, 39)
+## create a logical vector called college.aged
+## that is TRUE for someone between 18-22, inclusive
+## find the number of college.aged respondents
+```
+
+```{r ages-solution}
+college.aged <- (ages >= 18) & (ages <= 22)
+sum(college.aged)
+```
+
+```{r ages-check}
+grade_result_strict(
+  pass_if(~ identical(college.aged, (ages >= 18) & (ages <= 22))),
+  pass_if(~ identical(.result, sum((ages >= 18) & (ages <= 22))))
+)
+```
+
+
+### Subsetting based on logicals
+
+
+
+In the last exercise, you used logical statements to create a vector that told us whether each entry in the `ages` vector is in the 18-22 year-old range. We can now use that information to figure out what the actual ages of the respondents in that range are. 
+
+In base R, you can use the brackets for subsetting the data, and apply functions (such as `mean` to calculate the average) to this subset. For more details, please refer to the base R version of this tutorial.
+
+However, tidyverse functions for subsetting the data, such as `filter()` or `select()` cannot be applied to vectors. In this sense, if you would like to subset vectors, base R works better than tidyverse.
+
+
+### Using simple conditional statements
+
+For this exercise, we'll use the resume data once again.
+
+What if we wanted to create a new vector that depends on whether a statement is true or false? For example, suppose you wanted to create an indicator variable for whether or not a specific resume had the name "Carrie." We can  create a new variable using the `if_else(X, Y, Z)` command. This command takes a logical vector as `X` and returns a new vector of the same length as `X` that has the value `Y` if that value in `X` is TRUE and `Z` if that value in `X` is FALSE.
+
+
+#### Exercise
+
+- Use the `if_else` function to create a new variable called `carrie` that is 1 if the resume name (`firstname`) is `"Carrie"` and 0 otherwise.
```

Comment:
I think this is the first time that `mutate` is introduced. You might want to add a little guidance on how to use.

Reply:
I added a new exercise for `mutate` right before this one.